cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/RegulatingControlMappingForTransformers.java

```diff
@@ -0,0 +1,391 @@
+            throw new CgmesModelException("Transformer already added, Transformer id : " + transformerId);
+        }
+
+        RegulatingControlForTwoWindingsTransformer rc = new RegulatingControlForTwoWindingsTransformer();
+        rc.ratioTapChanger = rcRtc;
+        rc.phaseTapChanger = rcPtc;
+        t2xMapping.put(transformerId, rc);
+    }
+
+    public RegulatingControlRatio buildEmptyRegulatingControlRatio() {
+        RegulatingControlRatio rtc = new RegulatingControlRatio();
+        rtc.id = null;
+        rtc.regulatingControlId = null;
+        rtc.tculControlMode = null;
+        rtc.tapChangerControlEnabled = false;
+
+        return rtc;
+    }
+
+    public RegulatingControlRatio buildRegulatingControlRatio(String id, PropertyBag tc) {
+        String regulatingControlId = getRegulatingControlId(tc);
+        String tculControlMode = tc.get("tculControlMode");
+        boolean tapChangerControlEnabled = tc.asBoolean(TAP_CHANGER_CONTROL_ENABLED, false);
+        return buildRegulatingControlRatio(id, regulatingControlId, tculControlMode, tapChangerControlEnabled);
+    }
+
+    private RegulatingControlRatio buildRegulatingControlRatio(String id, String regulatingControlId,
+        String tculControlMode, boolean tapChangerControlEnabled) {
+        RegulatingControlRatio rtc = new RegulatingControlRatio();
+        rtc.id = id;
+        rtc.regulatingControlId = regulatingControlId;
+        rtc.tculControlMode = tculControlMode;
+        rtc.tapChangerControlEnabled = tapChangerControlEnabled;
+        return rtc;
+    }
+
+    public RegulatingControlPhase buildEmptyRegulatingControlPhase() {
+        RegulatingControlPhase rtc = new RegulatingControlPhase();
+        rtc.regulatingControlId = null;
+        rtc.tapChangerControlEnabled = false;
+
+        return rtc;
+    }
+
+    public RegulatingControlPhase buildRegulatingControlPhase(PropertyBag tc) {
+        String regulatingControlId = getRegulatingControlId(tc);
+        boolean tapChangerControlEnabled = tc.asBoolean(TAP_CHANGER_CONTROL_ENABLED, false);
+        boolean ltcFlag = tc.asBoolean("ltcFlag", false);
+        return buildRegulatingControlPhase(regulatingControlId, tapChangerControlEnabled, ltcFlag);
+    }
+
+    private RegulatingControlPhase buildRegulatingControlPhase(String regulatingControlId,
+        boolean tapChangerControlEnabled, boolean ltcFlag) {
+        RegulatingControlPhase rtc = new RegulatingControlPhase();
+        rtc.regulatingControlId = regulatingControlId;
+        rtc.tapChangerControlEnabled = tapChangerControlEnabled;
+        rtc.ltcFlag = ltcFlag;
+        return rtc;
+    }
+
+    public static void initializeRatioTapChanger(RatioTapChangerAdder adder) {
+        adder.setRegulationTerminal(null);
+        adder.setTargetV(Double.NaN);
+        adder.setTargetDeadband(Double.NaN);
+        adder.setRegulating(false);
+    }
+
+    public static void initializePhaseTapChanger(PhaseTapChangerAdder adder) {
```

Comment:
I don't think this method is useful as it is already the behavior of unset `PhaseTapChangerAdder`

Reply:
Deleted